src/main/java/org/thethingsnetwork/java/app/lib/Client.java

```diff
@@ -70,14 +72,34 @@
      * @param _broker The broker address, including protocol and port
      * @param _appId Your appId (or appEUI for staging)
      * @param _appAccessKey Your appAccessKey
+     * @throws java.net.URISyntaxException if the provided broker address is malformed
      */
-    public Client(String _broker, String _appId, String _appAccessKey) {
-        broker = _broker.contains(".") ? _broker : _broker + ".thethings.network";
+    public Client(String _broker, String _appId, String _appAccessKey) throws URISyntaxException {
+        broker = validateBroker(_broker);
         appId = _appId;
         connOpts.setUserName(_appId);
         connOpts.setPassword(_appAccessKey.toCharArray());
     }
 
+    private String validateBroker(String _source) throws URISyntaxException {
+
+        URI tempBroker = new URI(_source.contains(".") ? _source : (_source + ".thethings.network"));
+
+        switch (tempBroker.getScheme()) {
+            case "tcp":
+                if (tempBroker.getPort() == -1) {
+                    return tempBroker.toString() + ":1883";
+                }
+                break;
+            case "ssl":
```

Comment:
If it's SSL then probably you need certificates as well. How would you set that? Here's how I've set it up for Nodejs: https://github.com/TheThingsNetwork/node-app-lib/pull/20